### ISSUE TITLE
Cannot install r10k 3.0.0: 

### ISSUE THREAD

**Issue by mowtschan** [puppetlabs/r10k#855]
v2.6.4 works fine!
`gem install install --no-rdoc --no-ri r10k -v 2.6.4`

but v3.0.0 doesn't work
`gem install install --no-rdoc --no-ri r10k -v 3.0.0`

produces:
 ```
ERROR:  Error installing r10k:
	cri requires Ruby version ~> 2.3.
```

OS: Ubuntu 16.04.4 LTS
```
ruby -v
ruby 2.3.1p112 (2016-04-26) [x86_64-linux-gnu]
```
```
gem -v
2.2.5
```
```
puppet -V
4.10.6
```

**Comment by jonjhallettuob** [comment]
Also on CentOS 7: https://github.com/puppetlabs/r10k/issues/856

**Comment by MikaelSmith** [comment]
The CentOS 7 failure is expected (because it uses Ruby 2.0). I'm not sure what's going on with Ubuntu 16.04. With 16.04.3 I can install it just fine
```
Welcome to Ubuntu 16.04.3 LTS (GNU/Linux 4.4.0-21-generic x86_64)

 * Documentation:  https://help.ubuntu.com
 * Management:     https://landscape.canonical.com
 * Support:        https://ubuntu.com/advantage
root@gaq1y6ctdyfmkct:~# ls
linux.iso
root@gaq1y6ctdyfmkct:~# ruby --version
ruby 2.3.1p112 (2016-04-26) [x86_64-linux-gnu]
root@gaq1y6ctdyfmkct:~# gem install r10k
Fetching: colored-1.2.gem (100%)
Successfully installed colored-1.2
Fetching: cri-2.15.1.gem (100%)
Successfully installed cri-2.15.1
Fetching: log4r-1.1.10.gem (100%)
Successfully installed log4r-1.1.10
Fetching: multi_json-1.13.1.gem (100%)
Successfully installed multi_json-1.13.1
Fetching: multipart-post-2.0.0.gem (100%)
Successfully installed multipart-post-2.0.0
Fetching: faraday-0.13.1.gem (100%)
Successfully installed faraday-0.13.1
Fetching: faraday_middleware-0.12.2.gem (100%)
Successfully installed faraday_middleware-0.12.2
Fetching: semantic_puppet-1.0.2.gem (100%)
Successfully installed semantic_puppet-1.0.2
Fetching: minitar-0.6.1.gem (100%)
The `minitar` executable is no longer bundled with `minitar`. If you are
expecting this executable, make sure you also install `minitar-cli`.
Successfully installed minitar-0.6.1
Fetching: fast_gettext-1.1.2.gem (100%)
Successfully installed fast_gettext-1.1.2
Fetching: locale-2.1.2.gem (100%)
Successfully installed locale-2.1.2
Fetching: text-1.3.1.gem (100%)
Successfully installed text-1.3.1
Fetching: gettext-3.2.9.gem (100%)
Successfully installed gettext-3.2.9
Fetching: gettext-setup-0.30.gem (100%)
Successfully installed gettext-setup-0.30
Fetching: puppet_forge-2.2.9.gem (100%)
Successfully installed puppet_forge-2.2.9
Fetching: r10k-3.0.0.gem (100%)
Successfully installed r10k-3.0.0
Parsing documentation for colored-1.2
Installing ri documentation for colored-1.2
Parsing documentation for cri-2.15.1
Installing ri documentation for cri-2.15.1
Parsing documentation for log4r-1.1.10
Installing ri documentation for log4r-1.1.10
Parsing documentation for multi_json-1.13.1
Installing ri documentation for multi_json-1.13.1
invalid options: -SHN
(invalid options are ignored)
Parsing documentation for multipart-post-2.0.0
Installing ri documentation for multipart-post-2.0.0
Parsing documentation for faraday-0.13.1
Installing ri documentation for faraday-0.13.1
Parsing documentation for faraday_middleware-0.12.2
Installing ri documentation for faraday_middleware-0.12.2
Parsing documentation for semantic_puppet-1.0.2
Installing ri documentation for semantic_puppet-1.0.2
Parsing documentation for minitar-0.6.1
Installing ri documentation for minitar-0.6.1
Parsing documentation for fast_gettext-1.1.2
Installing ri documentation for fast_gettext-1.1.2
Parsing documentation for locale-2.1.2
Installing ri documentation for locale-2.1.2
Parsing documentation for text-1.3.1
Installing ri documentation for text-1.3.1
Parsing documentation for gettext-3.2.9
Installing ri documentation for gettext-3.2.9
Parsing documentation for gettext-setup-0.30
Installing ri documentation for gettext-setup-0.30
Parsing documentation for puppet_forge-2.2.9
Installing ri documentation for puppet_forge-2.2.9
Parsing documentation for r10k-3.0.0
Installing ri documentation for r10k-3.0.0
Done installing documentation for colored, cri, log4r, multi_json, multipart-post, faraday, faraday_middleware, semantic_puppet, minitar, fast_gettext, locale, text, gettext, gettext-setup, puppet_forge, r10k after 8 seconds
16 gems installed
```

Any chance you're installing to the Ruby version at /opt/puppetlabs/puppet/bin/ruby? For Puppet 4, I believe that's Ruby 2.1.

**Comment by mowtschan** [comment]
can you post your?:
`gem -v`

**Comment by MikaelSmith** [comment]
```
root@ku50lu8v0xiux71:~# gem -v
2.5.1
```

**Comment by mowtschan** [comment]
doesn't work with:  
```
/opt/puppetlabs/puppet/bin/gem -v
2.2.5
---
/opt/puppetlabs/puppet/bin/gem install r10k
ERROR:  Error installing r10k:
	cri requires Ruby version ~> 2.3.
```

but works with:
```
/usr/bin/gem -v               
2.5.2.1
---
/usr/bin/gem install r10k     
Successfully installed r10k-3.0.0
Parsing documentation for r10k-3.0.0
Installing ri documentation for r10k-3.0.0
Done installing documentation for r10k after 1 seconds
1 gem installed
```

**Comment by mowtschan** [comment]
you are right ruby inside of puppetlabs are 2.1:
```
/opt/puppetlabs/puppet/bin/ruby -v
ruby 2.1.9p490 (2016-03-30 revision 54437) [x86_64-linux]

```

**Comment by mowtschan** [comment]
How to fix this in a proper way? Upgrade puppet to v5.0?

**Comment by MikaelSmith** [comment]
Stick to r10k 2.x, upgrade to puppet 5, or use the system ruby and update gem (if that was the issue).

**Comment by mowtschan** [comment]
Thanks!

**Comment by zaxlofful** [comment]
I'm having this same issue on ALMA9, the included puppet version is 7.30.0; but the included Ruby version is much older (ruby 2.7.8p225).
